// emblem-brunch/lib/index.js
// Generated by CoffeeScript 1.4.0
(function() {
  var Emblem, EmblemCompiler, sysPath;

  Emblem = require('emblem');

  sysPath = require('path');

  module.exports = EmblemCompiler = (function() {

    EmblemCompiler.prototype.brunchPlugin = true;

    EmblemCompiler.prototype.type = 'template';

    EmblemCompiler.prototype.extension = 'emblem';

    EmblemCompiler.prototype.pattern = /\.(?:emblem)$/;

    function EmblemCompiler(config) {
      this.config = config;
      null;
    }

    EmblemCompiler.prototype.compile = function(data, path, callback) {
      var content, error, result;
      try {
        content = Emblem.precompile(require('handlebars'), data);
        return result = "module.exports = Handlebars.template(" + content + ");";
      } catch (err) {
        return error = err;
      } finally {
        callback(error, result);
      }
    };

    EmblemCompiler.prototype.include = [sysPath.join(__dirname, '..', 'vendor', 'handlebars.runtime.js')];

    return EmblemCompiler;

  })();

}).call(this);
